tests/integration/window-test.js

```diff
@@ -12,26 +12,38 @@ moduleForComponent('window', 'Integration | window', {
 });
 
 test('it can mock window in integration tests', async function(assert) {
+  window.localStorage.setItem('counter', '2');
 
   this.render(hbs`
     {{#window-tester as |window|}}
+      <span {{action window.increment}}>{{window.counter}}</span>
       <button {{action window.redirect "http://www.example.com"}}>Redirect</button>
     {{/window-tester}}
   `);
 
+  assert.equal(find('span').textContent.trim(), '2');
+
   await click('button');
 
   assert.equal(window.location.href, 'http://www.example.com/');
+
+  await click('span');
+
+  assert.equal(find('span').textContent.trim(), '3');
+  assert.equal(window.localStorage.getItem('counter'), '3');
 });
 
 test('each test gets a fresh copy - part 1 of 2', function(assert) {
   assert.notEqual(window.location.href, 'http://www.example.com/');
+  assert.notEqual(window.localStorage.getItem('counter'), '5');
```

Comment:
I think this should test for counter not being `'3'`, as this would be the its state from the previous test above, right?

Although I'm not sure we actually need these assertions here, as we test that state is not leaking in the following test as well...

Reply:
Ah, I misunderstood the meaning of the line above this one—I thought it was verifying that the value we were about to set was not already there.

I agree that the assertions are redundant, so I've removed both. Let me know if you'd rather I replace them instead.